src/lgfx/v1/platforms/portduino/Bus_SPI.cpp

```diff
@@ -0,0 +1,219 @@
+/*----------------------------------------------------------------------------/
+  Lovyan GFX - Graphics library for embedded devices.
+
+Original Source:
+ https://github.com/lovyan03/LovyanGFX/
+
+Licence:
+ [FreeBSD](https://github.com/lovyan03/LovyanGFX/blob/master/license.txt)
+
+Author:
+ [lovyan03](https://twitter.com/lovyan03)
+
+Contributors:
+ [ciniml](https://github.com/ciniml)
+ [mongonta0716](https://github.com/mongonta0716)
+ [tobozo](https://github.com/tobozo)
+/----------------------------------------------------------------------------*/
+#if defined (ESP_PLATFORM)
+#elif defined (ESP8266)
+#elif defined (__SAMD21__) || defined(__SAMD21G18A__) || defined(__SAMD21J18A__) || defined(__SAMD21E17A__) || defined(__SAMD21E18A__)
+#elif defined (__SAMD51__)
+#elif defined (STM32F2xx) || defined (STM32F4xx) || defined (STM32F7xx)
+#elif defined (ARDUINO_ARCH_SPRESENSE)
+#elif defined (ARDUINO_ARCH_MBED_RP2040) || defined(ARDUINO_ARCH_RP2040)
+#elif defined (PORTDUINO_LINUX_HARDWARE)
+
+#include "Bus_SPI.hpp"
+#include "../../misc/pixelcopy.hpp"
+#include <Arduino.h>
+#include <SPI.h>
+
+namespace lgfx
+{
+ inline namespace v1
+ {
+//----------------------------------------------------------------------------
+
+  void Bus_SPI::config(const config_t& config)
+  {
+    _cfg = config;
+
+    if (_cfg.pin_dc >= 0)
+    {
+      pinMode(_cfg.pin_dc, pin_mode_t::output);
+      gpio_hi(_cfg.pin_dc);
+    }
+  }
+
+  bool Bus_SPI::init(void)
+  {
+    dc_h();
+    pinMode(_cfg.pin_dc, pin_mode_t::output);
+    char x = (_cfg.spi_host & (1 << 4) - 1) + '0';
+    char y = (_cfg.spi_host >> 4) + '0';
+    PrivateSPI = new HardwareSPI();
```

Comment:
I would very much prefer to move the knowledge about X/Y into the lower layer, i.e. HardwareSPI, e.g. by adding a ctor HardwareSPI(uint8_t spi_host) and do the string assembly there. Also add a function overload for HardwareSPI::begin().

The result would be a much cleaner layering and free the graphics library from the task to assemble linux kernel specific device tree information. The graphics library would just transparently pass spi_host without the knowledge about its composition.

Reply:
And as already mentioned before, the variable _PrivateSPI_ has a poor naming and does not fit to the C++ naming conventions (C++ variables must not start with capital letter). I suggest to use _spi_ instead.

Reply:
> adding a ctor HardwareSPI(int8_t spi_host) and do the string assembly there

That would simplify this changeset quite a bit. In fact, it would eliminate the need for the added portduino platform here, I think. I'll work on it.